### PR TITLE
[WebNN EP] Use explicit padding

### PR DESCRIPTION
**Pull request by zesongw**
WebNN will remove autoPad option, we need to use explicit padding values.
Compute padding values of autopad(same-upper, same-lower) for Op Pool, Conv and ConvTranspose.

### CODE REVIEW

**Comment by zesongw** [comment]
@fdwr, @guschmue, PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc [hunk 1]
```diff
@@ -55,29 +55,85 @@ common::Status SetConvBaseOptions(ModelBuilder& model_builder,
...
-                                    auto_pad_type));
-  if (AutoPadType::SAME_UPPER == auto_pad_type || AutoPadType::SAME_LOWER == auto_pad_type) {
-    if (AutoPadType::SAME_LOWER == auto_pad_type) {  // default is SAME_UPPER
-      options.set("autoPad", emscripten::val("same-lower"));
+  AutoPadType auto_pad_type = StringToAutoPadType(helper.Get("auto_pad", "NOTSET"));
+  if (node.OpType() == "Conv") {
+    // Calculate explicit padding for autoPad.
+    if (AutoPadType::SAME_UPPER == auto_pad_type || AutoPadType::SAME_LOWER == auto_pad_type) {
+      std::vector<int64_t> pads_out;
+      ORT_RETURN_IF_ERROR(HandleAutoPad(input_shape, weight_shape[2], weight_shape[3],
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:65]: >weight_shape[2], weight_shape[3],

Do we verify somewhere else that the weight shape is at least 4 elements?

**Review by fdwr — APPROVED** [review]

File: onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc (hunk 1 above)
Comment (fdwr) [reply onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc:65]: Ah, checking that is in @honry's CR: https://github.com/microsoft/onnxruntime/pull/18703/files

**Comment by fdwr** [comment]
Huh, I'm not sure why the license/cla task is stalled, nor how to start it 🤨.

**Comment by fdwr** [comment]
@zesongw It looks like there are a few possibilities for the cla issue:
- Push another iteration: `git commit --allow-empty && git push`
- "...create a new PR for the same commit using different branch name. This will kick the build upon completion both original and new PR will get status update. Then you can merge the original and close the new one."  https://stackoverflow.com/a/54488712/937938
- Use the [CLA assistant](https://cla-assistant.io/) to "Recheck PR's" https://github.com/cla-assistant/cla-assistant/issues/124

**Pull request merged**